task8/src/main/java/PostServlet.java

```diff
@@ -8,80 +8,84 @@
+        response.setCharacterEncoding(ConstantDictionary.CHARACTER_ENCODING);
         if (post != null) {
             response.getWriter().write(post.toString());
         } else {
-            response.getWriter().write(RESULT_NOT_FOUND);
+            response.getWriter().write(ConstantDictionary.RESULT_NOT_FOUND);
         }
     }
 
     @Override
-    protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String str = request.getQueryString();
-        String sID = str.substring(3);
-        int id = Integer.parseInt(sID);
-        response.setContentType(CONTENT_TYPE);
-        response.setCharacterEncoding(CHARACTER_ENCODING);
-        if (posts.removePost(id)) {
-            response.getWriter().write(DELETE_POST_TEXT + id);
-            response.getWriter().write(RESULT_SUCCESS);
-        } else {
-            response.getWriter().write(RESULT_NOT_FOUND);
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response) {
+        String sID = request.getQueryString();
+        sID = sID.substring(3);
+        response.setContentType(ConstantDictionary.CONTENT_TYPE);
+        response.setCharacterEncoding(ConstantDictionary.CHARACTER_ENCODING);
+        try {
+            if (posts.removePost(sID)) {
+                response.getWriter().write(ConstantDictionary.DELETE_POST_TEXT + sID);
+                response.getWriter().write(ConstantDictionary.RESULT_SUCCESS);
+            } else {
+                response.getWriter().write(ConstantDictionary.RESULT_NOT_FOUND);
+            }
+        } catch (IOException e) {
+            log.log(Level.SEVERE, "Exception: ", e);
         }
     }
 
     @Override
-    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String author = request.getParameter(KEY_AUTHOR);
-        String description = request.getParameter(KEY_DESCRIPTION);
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) {
+        String author = request.getParameter(ConstantDictionary.KEY_AUTHOR);
+        String description = request.getParameter(ConstantDictionary.KEY_DESCRIPTION);
         Date creationDate = null;
         Date validateUntil = null;
         try {
-            creationDate = new SimpleDateFormat(DATE_PATTERN).parse(request.getParameter(KEY_CREATION_DATE));
-            validateUntil = new SimpleDateFormat(DATE_PATTERN).parse(request.getParameter(KEY_VALIDATE_UNTIL));
+            creationDate = new SimpleDateFormat(ConstantDictionary.DATE_PATTERN).parse(request.getParameter(ConstantDictionary.KEY_CREATION_DATE));
+            validateUntil = new SimpleDateFormat(ConstantDictionary.DATE_PATTERN).parse(request.getParameter(ConstantDictionary.KEY_VALIDATE_UNTIL));
         } catch (ParseException e) {
-            e.printStackTrace();
+            log.log(Level.SEVERE, "Exception: ", e);
```

Comment:
Usually, you write here an "explanation". It is already clear this is an exception) People try to describe in what circumstances it happened, what were the critical; values (like: "Exception when trying to request URL " + myUrl + " with query parameter " + param, etc). The rest will be (hopefully) clear from the `e.toString()` that is called under the hood